.github/workflows/link-checker.yaml

```diff
@@ -0,0 +1,44 @@
+name: Check Markdown links
+
+on:
+  pull_request:
+
+
+jobs:
+  markdown-link-check:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+
+    - name: Get all docs files that have changed
+      id: changed-files-yaml
+      uses: tj-actions/changed-files@v45
+      with:
+        files_yaml: |
+          doc:
+            - '**.md'
+            - docs/**
+
+    - name: Run step if doc file(s) change
+      if: steps.changed-files-yaml.outputs.doc_any_changed == 'true'
+      env:
+        DOC_ALL_CHANGED_FILES: ${{ steps.changed-files-yaml.outputs.doc_all_changed_files }}
+      run: |
+        echo "One or more doc file(s) has changed."
+        echo "List all the files that have changed: $DOC_ALL_CHANGED_FILES"
+
+    # If one file is changed, we can check all the files
+    - name: Check markdown links in docs
+      if: steps.changed-files-yaml.outputs.doc_any_changed == 'true'
+      uses: gaurav-nelson/github-action-markdown-link-check@v1
+      with:
+        config-file: '.github/workflows/link-checker.config.json'
+        folder-path: 'docs/'
```

Comment:
Does this mean that the markdown files at the root (like `README.md`) will not be checked?

Reply:
They are checked by the next step. It is a little bit hacky, but the reason is that we  don't want to check all the md files in the whole repository, as there are random md files that we don't control directly in this repo (like readme.md for static plugins etc...). 

It has been a while since I initially opened this so I need to check what is happening if I let it run on all md files

Reply:
There is a lot of broken links in other md files like in `./packages/app/CHANGELOG.md` (see. https://github.com/janus-idp/backstage-showcase/actions/runs/11859232155/job/33051687669?pr=1779) 

I can fix them, but I'm not sure if going back to Changelog and fixing no longer existing links makes sense. I consider Changelog as a snapshot in a given time, not sure if going back and updating or removing links makes sense